src/query/plan/rule_based_planner.hpp

```diff
@@ -530,6 +535,28 @@ class RuleBasedPlanner {
     }
     return std::make_unique<plan::Merge>(std::move(input_op), std::move(on_match), std::move(on_create));
   }
+
+  void HandleForeachClause(query::Foreach *foreach, std::unique_ptr<LogicalOperator> &input_op,
```

Comment:
I don't like that `input_op` is an out parameter. Why don't the function has a return value instead?